pkg/kubelet/container/types.go

```diff
@@ -0,0 +1,173 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package container
+
+import (
+	"errors"
+	"io"
+	"time"
+
+	"github.com/fsouza/go-dockerclient"
+)
+
+var (
+	ErrNoSuchImage = errors.New("no such image")
+)
+
+// Container represents a container.
+type Container struct {
+	ID              string            `json:"id"`
+	Name            string            `json:"name,omitempty"`
+	Names           []string          `json:"names,omitempty"`
+	Image           string            `json:"image,omitempty"`
+	ImageID         string            `json:"imageID`
+	Command         string            `json:"command,omitempty"`
+	Created         time.Time         `json:"created,omitempty"`
+	State           State             `json:"state,omitempty"`
+	Status          string            `json:"status,omitempty"`
+	NetworkSettings *NetworkSettings  `json:"networkSettings`
+	Ports           []Port            `json:"ports,omitempty"`
+	SizeRw          int64             `json:"sizeRw,omitempty"`
+	SizeRootFs      int64             `json:"sizeRootFs,omitempty"`
+	Volumes         map[string]string `json:"volumes,omitempty"`
+}
+
+// Port is a type that represents a port mapping.
+type Port struct {
+	PrivatePort int64  `json:"privatePort,omitempty"`
+	PublicPort  int64  `json:"publicPort,omitempty"`
+	Type        string `json:"type,omitempty"`
+	IP          string `json:"ip,omitempty"`
+}
+
+// State represents the state of a container.
+type State struct {
+	Running    bool      `json:"running,omitempty"`
+	Paused     bool      `json:"paused,omitempty"`
+	OOMKilled  bool      `json:"oomkilled,omitempty"`
+	Pid        int       `json:"pid,omitempty"`
+	ExitCode   int       `json:"exitCode,omitempty"`
+	Error      string    `json:"error,omitempty"`
+	StartedAt  time.Time `json:"startedAt,omitempty"`
+	FinishedAt time.Time `json:"finishedAt,omitempty"`
+}
+
+// NetworkSettings contains network-related information about a container.
+type NetworkSettings struct {
+	IPAddress string `json:"ipAddress,omitempty"`
+}
+
+// HostConfig contains the container options related to starting a container on a given host.
+type HostConfig struct {
+	Binds        []string                 `json:"binds,omitempty"`
+	CapAdd       []string                 `json:"capAdd,omitempty"`
+	CapDrop      []string                 `json:"capDrop,omitempty"`
+	Privileged   bool                     `json:"privileged,omitempty"`
+	PortBindings map[string][]PortBinding `json:"portBindings,omitempty"`
+	DNS          []string                 `json:"dns,omitempty"`
+	DNSSearch    []string                 `json:"dnsSearch,omitempty"`
+	NetworkMode  string                   `json:"networkMode,omitempty"`
+	IpcMode      string                   `json:"ipcMode,omitempty"`
+}
+
+// PortBinding represents the host/container port mapping.
+type PortBinding struct {
```

Comment:
Again, I know this is more or less direct from docker, but can we clean this up?  What is the meaning of Port vs PortBinding, for instance.

If we now own this abstraction layer, can we make it something that makes sense?